### PR TITLE
Perform search indexing dry run as part of PR checks

### PR DESCRIPTION
**Pull request by florkbr**
See https://issues.redhat.com/browse/RHCLOUD-27749

### CODE REVIEW

**Comment by florkbr** [comment]
Sample output of the new make command locally:
```bash
➜  chrome-service-backend git:(improve-search-pr-checks) make publish-search-index-dry-run
go run cmd/search/*
Publishing search index
Attempt dry run search index for stage environment.
[ERROR] Failed to deploy search index for stage environment.
[ERROR] parsing link for href entry at static/stable/stage/navigation/iam-navigation.json.navItems[3]
Attempt dry run search index for prod environment.
[WARN] Unable to convert link id  to string. map[appId:patch href:/insights/patch/systems product:Red Hat Insights title:Systems] in file static/stable/prod/navigation/insights-navigation.json.navItems[2].routes
[WARN] Unable to convert link id  to string. map[appId:patch href:/insights/patch/templates product:Red Hat Insights title:Templates] in file static/stable/prod/navigation/insights-navigation.json.navItems[2].routes
[WARN] Unable to convert link id  to string. map[appId:drift href:/insights/drift/ product:Red Hat Insights title:Comparison] in file static/stable/prod/navigation/insights-navigation.json.navItems[3].routes
[WARN] Unable to convert link id  to string. map[appId:drift href:/insights/drift/baselines product:Red Hat Insights title:Baselines] in file static/stable/prod/navigation/insights-navigation.json.navItems[3].routes
[ERROR] Failed to deploy search index for prod environment.
[ERROR] parsing link for href entry at static/stable/prod/navigation/insights-navigation.json.navItems[4]
Search index publishing failed. See above errors.
exit status 1
make: *** [publish-search-index-dry-run] Error 1
```

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -67,3 +73,6 @@ jobs:
 
       - name: Test
         run: make test
+      
+      - name: Search index dry run
```
Comment (Hyperkid123) [inline .github/workflows/main.yaml:77]: We want to run this as a pr_check. This script should be added to the `Dockerfile` after the `make validate-schema` to block the Jenkins build and failing the checks/build.

Comment (florkbr) [reply .github/workflows/main.yaml]: Alright. My logic here was that the PR check would be a bit more concise as to why the PR is failing (versus having to dive into Jenkins and interpret the output). 

For now I stuck it in the Dockerfile. If it looks good to you I'll fix the various issues so this PR passes.

Comment (Hyperkid123) [reply .github/workflows/main.yaml]: Yup sounds good.

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**